docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+export function NetworkedFooPrefab(params: NetworkedFooPrefabParams) {
+  return (
+    <entity
+      networked
+      foo={{ val: params.val }}
+    />
+  );
+}
 
-For the most part, users of the networking systems do not need to understand these concepts. These are handled internally by the systems themselves. However, users do need to understand that ownership is not transactional or guaranteed. That is, ownership is not &ldquo;requested and then transferred&rdquo;, and just because one client claims ownership of an entity does not mean that other clients will respect that claim.
+prefabs.set("networked-foo", { template: NetworkedFooPrefab } );
+```
 
-Users can inspect the state the `Networked` or `Owned` components as needed in cases when their ownership claims matter. They may find themselves writing coroutines that looks like this:
+And then write a network schema which defines how component data is packed in
+network message and register it in the built-in [`schemas`](#network-schema)
+map.
 
 ```typescript
-takeSoftOwnership(world, eid);
-yield sleep( 3000 ); // Wait a few seconds to see if we "win" ownership
-if (!hasComponent(world, Owned, eid)) return;
+// src/network-schemas/foo.ts
+
+import { Foo } from "../components/foo";
+import { schemas } from "../utils/network-schemas";
+import { defineNetworkSchema } from "../utils/define-network-schema";
+
+const runtimeSerde = defineNetworkSchema(Foo);
+export const NetworkedFoo: NetworkSchema = {
+  componentName: "foo",
+  serialize: runtimeSerde.serialize,
+  deserialize: runtimeSerde.deserialize
+};
+
+schemas.set(NetworkedFoo, NetworkedFooSchema);
 ```
 
-If this becomes a common and error-prone pattern, then we may introduce helper functions or additional semantics to cover these cases.
+Now, `Foo` component data is ready for network sync. The built-in
+[`createNetworkedEntity()`](#createnetworkedentity) function locally creates
+new entities with associated components set up, and also sends a message to
+remote clients to cause the same entity and components set up there. When `Foo`
+component data is updated, Hubs Client sends a message to remote Hubs Clients
+for sync.
 
+```typescript
+import { createNetworkedEntity } from "../utils/create-networked-entity";
+
+const eid = createNetworkedEntity(world, "networked-foo", { val: 0 });
+```
+
+Let's dive into more details below.
+ 
 
-<a id="org62cc95d"></a>
+## Creating Networked Entities
 
-# Creating Networked Entities
+`Networked entities` are any entities with the the built-in `Networked`
+component. Hubs Clients make their components data synched with remote clients.
 
-Client code creates networked entities by calling `createNetworkedEntity`, passing:
 
--   A `prefabName`, to indicate which prefab to initialize,
--   An `initialData` struct, to know how to initialize the prefab.
+### createNetworkedEntity()

```

Comment:
Should this section come after the "Prefab" section since it is dependent on it?

Reply:
Updated, thanks.